### ISSUE TITLE
Profiling Django with DEBUG=False does not work

### ISSUE THREAD

**Issue by divick** [pympler/pympler#43]
The docs mention usage of pympler with Django debug toolbar but debug toolbar does not work in non DEBUG mode. The Django Debug toolbar documentation clearly says that Debug toolbar panels will not show in non DEBUG mode. Moreover Django documentation mentions to set DEBUG=False in production otherwise the memory usage keeps increasing as all queries are kept in memory. In such a case how does one identify memory leaks in Django with pympler.

**Comment by Pankrat** [comment]
I usually do runtime or memory profiling on production-like staging machines where it is acceptable to activate debug mode. IMHO the kind of profiling that pympler does is too intrusive for a production environment. 

You've got a good point that activating DEBUG in Django will change the way memory is freed so you might chase leaks that only appear in DEBUG mode. This hasn't been a problem for my use-cases but it could affect yours.

As an alternative, you could also add a custom middleware and use one of the Pympler facilities to profile your request. The Django Debug Toolbar is mainly used for presenting the profiling results but you could also log the interesting information to the console from your middleware.

Does this answer your question?

Sorry for the late reply.

**Comment by divick** [comment]
@Pankrat Thanks for the reply. So if I understand correctly, it is not meant to work when DEBUG=False? If it is by design, then I guess mentioning so in the documentation would be useful. If it is not by design, then when DEBUG=False, it would be worthwhile to make it work, so that even in production environments, one can dump some statistics.

**Comment by Pankrat** [comment]
Yes, it's by design. A mention in the documentation sounds like a good idea. I'll keep this issue open until that happens.
